src/go/pt-galera-log-explainer/display/timelinecli.go

```diff
@@ -0,0 +1,352 @@
+	numberFound       int
+}
+
+// 4 here means there are 4 rows to store
+// 0: base info, 1: type of info that changed, 2: just an arrow placeholder, 3: new info
+const RowPerTransitions = 4
+
+type transitionSummary [RowPerTransitions]string
+
+// because only those transitions are implemented: file path, ip, node name, version
+const NumberOfPossibleTransition = 4
+
+// transactionSeparator is useful to highlight a change of context
+// example, changing file
+//   mysqld.log.2
+//    (file path)
+//           V
+//   mysqld.log.1
+// or a change of ip, node name, ...
+// This feels complicated: it is
+// It was made difficult because of how "tabwriter" works
+// it needs an element on each columns so that we don't break columns
+// The rows can't have a variable count of elements: it has to be strictly identical each time
+// so the whole next functions are here to ensure it takes minimal spaces, while giving context and preserving columns
+func transitionSeparator(keys []string, oldctxs, ctxs map[string]types.LogCtx) string {
+
+	ts := map[string]*transitions{}
+
+	// For each columns to print, we build tests
+	for _, node := range keys {
+		ctx, ok1 := ctxs[node]
+		oldctx, ok2 := oldctxs[node]
+
+		ts[node] = &transitions{tests: []*transition{}}
+		if ok1 && ok2 {
+			ts[node].tests = append(ts[node].tests, &transition{s1: oldctx.FilePath, s2: ctx.FilePath, changeType: "file path"})
+
+			if len(oldctx.OwnNames) > 0 && len(ctx.OwnNames) > 0 {
+				ts[node].tests = append(ts[node].tests, &transition{s1: oldctx.OwnNames[len(oldctx.OwnNames)-1], s2: ctx.OwnNames[len(ctx.OwnNames)-1], changeType: "node name"})
+			}
+			if len(oldctx.OwnIPs) > 0 && len(ctx.OwnIPs) > 0 {
+				ts[node].tests = append(ts[node].tests, &transition{s1: oldctx.OwnIPs[len(oldctx.OwnIPs)-1], s2: ctx.OwnIPs[len(ctx.OwnIPs)-1], changeType: "node ip"})
+			}
+			if oldctx.Version != "" && ctx.Version != "" {
+				ts[node].tests = append(ts[node].tests, &transition{s1: oldctx.Version, s2: ctx.Version, changeType: "version"})
+			}
+
+		}
+
+		// we resolve tests
+		ts[node].fillEmptyTransition()
+		ts[node].iterate()
+	}
+
+	highestStackOfTransitions := 0
+
+	// we need to know the maximum height to print
+	for _, node := range keys {
+		if ts[node].numberFound > highestStackOfTransitions {
+			highestStackOfTransitions = ts[node].numberFound
+		}
+	}
+	// now we have the height, we compile the stack to print (possibly empty placeholders for some columns)
+	for _, node := range keys {
+		ts[node].stackPrioritizeFound(highestStackOfTransitions)
+	}
+
+	out := "\t"
+	for i := 0; i < highestStackOfTransitions; i++ {
+		for row := 0; row < RowPerTransitions; row++ {
+			for _, node := range keys {
+				out += ts[node].transitionToPrint[i].summary[row]
+			}
+			if !(i == highestStackOfTransitions-1 && row == RowPerTransitions-1) { // unless last row
+				out += "\n\t"
+			}
+		}
+	}
+
+	if out == "\t" {
+		return ""
+	}
+	return out
+}
+
+func (ts *transitions) iterate() {
```

Comment:
It looks like this method counts something, probably is should have a different name.

Reply:
"iterate" was named that way to help readability in the main "transitionSeparator" func
I would vote to keep it that way